Update guide for Camunda exporter

ChrisKujawa opened the pull request:
## Description

This PR adds documentation and details about the Camunda Exporter and the related update guide.

Cover:

 * prefix migration
 * importer deployment and dependencies to Exporter
 * migration application
 * When to shutdown importers, etc.

closes https://github.com/camunda/camunda-docs/issues/4498
closes https://github.com/camunda/camunda-docs/issues/4493
closes https://github.com/camunda/camunda-docs/pull/4844

<!-- Provide an overview of what to expect in the PR. -->
<!-- Relate or link the associated epic or task. -->
<!-- Add `@camunda/tech-writers` as reviewer to pull in a tech writer, or add your embedded tech writer. -->

## When should this change go live?

<!-- PRs merged go to stage.docs.camunda.io first and must be manually released to docs.camunda.io. -->

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**. (add `bug` or `support` label)
- [ ] This is already available but undocumented and should be released within a week. (add `available & undocumented` label)
- [x] This is on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (create draft PR and/or add `hold` label)
- [x] This is part of a scheduled **alpha or minor**. (add alpha or minor label)
- [ ] There is **no urgency** with this change (add `low prio` label)

## PR Checklist

<!-- Camunda maintains 18 months of minor versions. Backporting your change to multiple versions is common. -->

- [ ] My changes are for **an upcoming minor release** and:
  - [x] are in the `/docs` directory (version 8.8).
  - [ ] are in the `/versioned_docs/version-8.7/` directory (version 8.7).
- [ ] My changes are for an **already released minor** and are in a `/versioned_docs` directory.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Adding or removing pages requires extra steps.
- [ ] I included my new page in the sidebar file(s).
- [ ] I added a redirect for a renamed or deleted page to the .htaccess file.
-->

- [ ] I added a DRI, team, or delegate as a reviewer for technical accuracy and grammar/style:
  - [ ] [Engineering team review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process)
  - [x] [Technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process) via `@camunda/tech-writers` unless working with an embedded writer.

<!-- UNCOMMENT THIS SECTION IF APPLICABLE. Changes to **docs infra**, including updates to workflows and adding new npm packages, must be first discussed via issue or #ask-c8-documentation and linked for context.
- [ ] My changes require a [docs infrastructure review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process). (add `dx` label) -->

akeller commented:
Before reviewing, I fixed your build issue and added the deploy label. I've also added tech writers for a more extensive review.

aleksander-dytko commented on docs/reference/announcements/880.md:
```diff
@@ -123,25 +123,20 @@ Additional upgrade considerations are necessary for deployments that use custom
-- The archiver, which takes care of the archiving of completed process instances, will be moved into the Zeebe system as well, to reduce the installation complexity and provide a better scaling and replication factor (based on partitions).
-- This helps achieve a streamlined architecture, and improves platform performance and stability (especially regarding ES/OS).
-- A new separate component covers the migration, which will be part of the single application but can also deployed separately. It will adjust the previous Operate indices to make them more harmonized and usable by Tasklist.
+The new Camunda Exporter helps us achieve a more streamlined architecture, better performance, and stability (especially concerning ES/OS). The target release for the Camunda Exporter project is the 8.8 release.
```
```suggestion
The new Camunda Exporter helps us achieve a more streamlined architecture, better performance, and stability (especially concerning ES/OS).
```
We have this in 8.8 release announcements so no need to highlight that

aleksander-dytko commented on docs/reference/announcements/880.md:
```diff
@@ -123,25 +123,20 @@ Additional upgrade considerations are necessary for deployments that use custom
 ### Harmonized index schema
 
-Camunda is harmonizing our index structure and usage.
+Our existing data schema in the secondary storage has been harmonized, to be used by all our components.
```
We need to add documentation about `primary` and `secondary storage` in C8. @akeller would your team be able to help with that?

akeller replied:
Is there a specific epic tied to an alpha? @conceptualshark might be able to assist, but would need to partner with @ChrisKujawa or someone else involved with the project and have capacity. What kind of timeline are we looking at here?

aleksander-dytko replied:
These docs won't be specifically related to the exporter topic, it's more of a general concept that we can refer to from exporter or rdbms topics. I think it would be good to pair with @romansmirnov to provide a clear explanation of these concepts. 
Would someone from your team be able to lead this effort?

akeller replied:
@conceptualshark do you have capacity for this? Delivered with 8.8, but might need a skeleton structure page in an upcoming alpha. Regardless, we'll need a docs issue for this since it won't be covered by either epic directly, but could be linked to both.

conceptualshark replied:
@akeller I can definitely take this on - I made a [placeholder issue](https://github.com/camunda/documentation-team/issues/458), but would love some input on filling it out. I can link it to this issue, but are there other epics that it should relate to?

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+We were able to reduce the need for several indices and to make use of existing ones (Operate indices are used by Tasklist, etc.)
+
+### Notes for updating to 8.7
```
```suggestion
### Notes for updating to 8.8
```

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+As we mentioned above we need to import old data from 8.7 via the importers.
+The importers need to run first and complete before the Camunda Exporter can confidently start.
+
+The Importer is reading data, from existing Zeebe Elasticsearch indices, and as soon it detects data from newer versions, like 8.8
```
```suggestion
The Importer is reading data, from existing Zeebe Elasticsearch indices, and as soon it detects data from newer versions,
```

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+it marks itself as complete. The Camunda Exporter can detect this and will start exporting only if the Importer is done.
+
+:::important
+To reduce the here potential of building a backlog, which can cause, on the Zeebe side, issues
```
```suggestion
To reduce the potential of building a backlog,
```

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+:::important
+To reduce the here potential of building a backlog, which can cause, on the Zeebe side, issues
+(like growing disk, as records don't get exported and cleaned up, etc.), the user need to make sure to drain the backlog queue before the update.
```
```suggestion
 the user needs to make sure to drain the backlog queue as much as possible before the update.
```

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+As we have harmonized indices there are some details Tasklist needs as part of the processes, which were not part of
+existing Operate process indices.
+
+For such we have built a migration application to enhance existing process models. This migration application will
```
```suggestion
For this reason, we have built a migration application to enhance existing process models. This migration application will
```

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+existing Operate process indices.
+
+For such we have built a migration application to enhance existing process models. This migration application will
+run asynchronous next to the other applications, and until the importers are done.
```
```suggestion
run asynchronous next to the other applications, until the importers are done.
```

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+:::warning
+Disk usage will massively increase while the migration is ongoing.
+Make sure you have at least double of disk space free.
```
Can we add that it will return to "normal" state after migration?

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+:::
+
+:::note
+While we test and validate the provided migration script.
```
```suggestion
While we test and validate the provided migration script,
```

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+:::note
+While we test and validate the provided migration script.
+It is recommended that you first back up the Elasticsearch, and test the migration script in a test environment first
```
```suggestion
it is recommended that you first back up the Elasticsearch, and test the migration script in a test environment first
```

aleksander-dytko commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+:::
+
+:::note
+The prefix migration involves downtime.
```
Can we add some explanation of what it means for users and that processing is not affected?

EuroLew commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+Without these configuration changes the update to 8.8 will fail.
+:::
+
+Given a new desired prefix of `some-prefix`
```
shouldn't this say `some-new-prefix`

aabouzaid commented on docs/reference/announcements/880.md:
```diff
@@ -123,25 +123,20 @@ Additional upgrade considerations are necessary for deployments that use custom
+Our web applications have historically used importers and archivers to consume, aggregate, and archive historical data provided by our Elasticsearch (ES) or OpenSearch (OS) Exporter.
 
-The exporter can consume Zeebe records (mostly events created by the engine), aggregate data, and store the related data into shared and harmonized indices.
+![current-state](../img/current-state-refined.png)
```
I'd avoid using `current` as it doesn't refer to which version it is.

ChrisKujawa replied:
I dont see it as an issue. It is in an announcement state for the 8.8 release. But for sure it can be renamed as well.

ChrisKujawa replied:
Done.

aabouzaid commented on docs/reference/announcements/880.md:
```diff
@@ -123,25 +123,20 @@ Additional upgrade considerations are necessary for deployments that use custom
+In the past release, we’ve engineered a new Camunda Exporter, which brings the importer and archiving logic of web components (Tasklist and Operate) closer to our distributed platform (Zeebe). This allows us to simplify our installation, enable scalability for web apps, reduce the latency to show runtime and historical data, and reduce data duplication (resource consumption).
 
-The following diagram shows a simplified version of this work.
+![target](../img/target-brown-field-refined-without-optimize.png)
```
I'd suggest using `green/brown` for terminology (only `brown` is shown in the PR).

aabouzaid commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+the user needs to make sure to drain the backlog queue as much as possible before the update.
+:::
+
+The backlog can be validated via the following metrics:
```
I suggest to make it explicit:

`following metrics` => `following Prometheus metrics`

aabouzaid commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+Steps for prefix migration:
+
+1. Shut down your Camunda 8.7 cluster, fex. reduce the replicas to zero. Such that no traffic will executed against your to-be-migrated ES/OS prefixes.
+2. In the `camunda/bin` directory there is a `prefix-migration` script that will execute the prefix migration,
```
Where is that instance running if the cluster is already shut down?

ChrisKujawa replied:
Do you mean the `prefix-migration` application? Can be everywhere, locally or deployed. But in this example locally I would suspect.

aabouzaid replied:
In that case, let's mention it explicitly to avoid any confusion from the reader.

ChrisKujawa replied:
I did in my last commit. :)

ChrisKujawa replied:
https://github.com/camunda/camunda-docs/pull/5023/commits/3040ddde6e48b683486300cced3aaea6366ba156

ChrisKujawa commented:
Hey @camunda/tech-writers how can we proceed with this PR and get this merged? I feel like we losing a lot of momentum. I think we can iterate and improve over the content as well after merging it. There doesn't seem to be a big blocker in my opinion.

ChrisKujawa commented:
@aleksander-dytko can you take another look and approve if possible so we can get this merged?

ChrisKujawa commented:
Maybe @RomanJRW might be also want to look at it

RomanJRW commented on docs/reference/announcements/880.md:
```diff
@@ -137,25 +137,20 @@ Additional upgrade considerations are necessary for deployments that use custom
 ## Camunda Exporter <span class="badge badge--long" title="This feature affects Self-Managed">Self-Managed</span>
 
-A new Camunda Exporter brings the importer and archiving logic of web components (Tasklist and Operate) closer to the distributed platform (Zeebe). The index schema is also being harmonized.
+Our web applications have historically used importers and archivers to consume, aggregate, and archive historical data provided by our Elasticsearch (ES) or OpenSearch (OS) Exporter.
```
```suggestion
Our web applications have historically used importers and archivers to consume, aggregate, and archive historical data provided by our Elasticsearch (ES) or OpenSearch (OS) Exporters.
```
Slightly ashamed as a native speaker that I am not fully sure if my suggestion is correct here. Maybe the tech writers can confirm 🙈

RomanJRW commented on docs/reference/announcements/880.md:
```diff
@@ -137,25 +137,20 @@ Additional upgrade considerations are necessary for deployments that use custom
 
-- Data is archived in the background, coupled to the exporter but without blocking the exporter's progress.
-- Indices can be located in either Elasticsearch (ES) or OpenSearch (OS). Our web components (Tasklist and Operate) then use the new harmonized indices to show data to the user.
+In the past release, we’ve engineered a new Camunda Exporter, which brings the importer and archiving logic of web components (Tasklist and Operate) closer to our distributed platform (Zeebe). This allows us to simplify our installation, enable scalability for web apps, reduce the latency to show runtime and historical data, and reduce data duplication (resource consumption).
```
```suggestion
In the past release, we’ve engineered a new Camunda Exporter, which brings the importing and archiving logic of web components (Tasklist and Operate) closer to our distributed platform (Zeebe). This allows us to simplify our installation, enable scalability for web apps, reduce the latency to show runtime and historical data, and reduce data duplication (resource consumption).
```

RomanJRW commented on docs/reference/announcements/880.md:
```diff
@@ -137,25 +137,20 @@ Additional upgrade considerations are necessary for deployments that use custom
-- The archiver, which takes care of the archiving of completed process instances, will be moved into the Zeebe system as well, to reduce the installation complexity and provide a better scaling and replication factor (based on partitions).
-- This helps achieve a streamlined architecture, and improves platform performance and stability (especially regarding ES/OS).
-- A new separate component covers the migration, which will be part of the single application but can also deployed separately. It will adjust the previous Operate indices to make them more harmonized and usable by Tasklist.
+The new Camunda Exporter helps us achieve a more streamlined architecture, better performance, and stability (especially concerning ES/OS).
```
```suggestion
The new Camunda Exporter helps us achieve a more streamlined architecture, better performance, and improved stability (especially concerning ES/OS).
```
I think it's missing an adjective, "improved" being just one possible example

RomanJRW commented on docs/reference/announcements/880.md:
```diff
@@ -137,25 +137,20 @@ Additional upgrade considerations are necessary for deployments that use custom
+For more details about this project, you can also read our blog post [here](https://camunda.com/blog/2025/02/one-exporter-to-rule-them-all-exploring-camunda-exporter/).
 
 ### Harmonized index schema
 
-Camunda is harmonizing our index structure and usage.
+Our existing data schema in the secondary storage has been harmonized, to be used by all our components.
```
Do users really know what harmonized indices mean? I feel like we should be a bit more explicit in saying exactly what we are referring to here

ChrisKujawa replied:
This terminology is only used for the announcement page, to describe also the process of us harmonizing the indices, afterwards we don't want to use it anymore

RomanJRW commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+![87-state](./img/87-state-refined.png)
+
+Our web applications have historically used importers and archivers to consume, aggregate, and archive historical data provided by our Elasticsearch (ES) or OpenSearch (OS) Exporter.
+With the introduction of the Camunda Exporter, there are some changes in the architecture users should be aware of.
```
same comment here (and elsewhere) as in the 8.8 announcements above

RomanJRW commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+The harmonized schema looks like the following:
+
+![harmoninzed-schema](./img/harmonized-indices-miro-v2.png)
```
```suggestion
![harmonized-schema](./img/harmonized-indices-miro-v2.png)
```
Small typo

RomanJRW commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+### Notes for updating to 8.8
+
+Notes for updating to 8.7 related to the Camunda Exporter and harmonized indices.
```
```suggestion
Notes for updating to 8.8 related to the Camunda Exporter and harmonized indices.
```

RomanJRW commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+![harmoninzed-schema](./img/harmonized-indices-miro-v2.png)
+
+We were able to reduce the need for several indices and to make use of existing ones (Operate indices are used by Tasklist, etc.)
```
Again, a more user-friendly description of what is meant by the harmonized indices would be nice

RomanJRW commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+existing Operate process indices.
+
+For this reason, we have built a migration application to enhance existing process models. This migration application will
+run asynchronous next to the other applications, until the importers are done.
```
```suggestion
run asynchronously next to the other applications, until the importers are done.
```

RomanJRW commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,152 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+:::note
+While we test and validate the provided migration script,
+it is recommended that you first back up the Elasticsearch, and test the migration script in a test environment first
```
```suggestion
it is recommended that you first back up the Database, and test the migration script in a test environment first
```

RomanJRW reviewed (approved):
Generally, this is a nice starting point upon which we can iterate moving forward. I have left some general comments but it would be good to have a tech writer go over some of this in more detail to make sure there is sufficient clarity with some of the terminology and instructions we are documenting here 👍

Pre-approving anyway, as I don't think my comments are blockers

aabouzaid reviewed (dismissed):
@ChrisKujawa All looks good, thanks for your effort :clap:

akeller commented:
> Hey @camunda/tech-writers how can we proceed with this PR and get this merged? I feel like we losing a lot of momentum. I think we can iterate and improve over the content as well after merging it. There doesn't seem to be a big blocker in my opinion.

@conceptualshark can you review this with the intention of helping the team incrementally deliver? Or tag in another tech writer to review?

conceptualshark commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,140 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
 
+## Camunda Exporter
+
+The new Camunda Exporter brings the importing and archiving logic of web components (Tasklist and Operate) closer to the distributed platform (Zeebe). This simplifies the installation, enables scalability for the web applications, reduces latency when showing runtime and historical data, and reduces data duplication (resource consumption).
+
+![87-state](./img/87-state-refined.png)
+
+Camunda web applications use importers and archivers to consume, aggregate, and archive historical data provided by the Elasticsearch (ES) or OpenSearch (OS) exporters. The new Camunda Exporter introduces changes in the architecture users should be aware of.
+
+During the migration/update phase, the Operate and Tasklist importer and the Elasticsearch exporter are still required. This allows 8.7 data written to the Zeebe journal, or already in the Zeebe Elasticsearch indices, to be imported into the new harmonized index schema.
+
+![brownfield](./img/target-brown-field-refined.png)
+
+### Harmonized Indices
+
+The harmonized schema looks like the following:
+
+![harmonized-schema](./img/harmonized-indices-miro-v2.png)
+
+This reduces the need for several indices and to make use of existing ones (Operate indices are used by Tasklist, etc.)
```
Since the content is largely the same, I think it would be appropriate to just link directly to the announcement here, and that way this guide can only include the upgrade notes. This would allow you to increase the headers on the later content and make it a bit easier to browse.

conceptualshark commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,140 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+- `operate_import_time_seconds_sum`
+- `operate_import_time_seconds_count`
+
+For example, the following query shows the import latency. If the result is in a range of seconds, it is fine to update.
```
Can we be more specific with a recommendation here? Is "a range of seconds" anything under a minute? 

```suggestion
For example, the following query shows the import latency. We recommend updating only if the result is within [recommended range of seconds].
```

conceptualshark commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,140 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+#### Ensure the exporter/import backlog is small
+
+For data from 8.7 to be imported, the importers must run first and complete before the Camunda Exporter can start.
+
+The importer reads data from existing Zeebe Elasticsearch indices, and when it detects data from newer versions, it marks itself as complete. The Camunda Exporter then starts exporting only when the importer is done.
+
+:::important
+To reduce the potential of building a backlog,
+the user needs to make sure to drain the backlog queue as much as possible before the update.
+:::
```
Not necessary here, but since the header for this section is on backlog size, it would be nice to know what would happen if the backlog is too large and you attempt to upgrade anyway. 

```suggestion
For data from 8.7 to be imported, the importers must first run and complete before the Camunda Exporter can start. The importer reads data from existing Zeebe Elasticsearch indices, and when it detects data from newer versions, it marks itself as complete. The Camunda Exporter then starts exporting only when the importer is done.

If the import backlog is too large, [why do we require the backlog to be small].

:::important
To reduce the potential of building a backlog, drain the backlog queue as much as possible before the update.
:::
```

conceptualshark commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,140 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+#### Data migration
+
+To enable Tasklist to utilize the new harmonized indices, Camunda 8.8 provides a migration application to enhance existing process models. This migration application runs asynchronously to the other applications until the importers are done, and ensures that all processes created in 8.7 are migrated and enhanced.
```
I reworded this a little bit but want to make sure the content is still correct - just calling it out for review!

ChrisKujawa replied:
Looks good thanks

conceptualshark commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,140 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+To enable Tasklist to utilize the new harmonized indices, Camunda 8.8 provides a migration application to enhance existing process models. This migration application runs asynchronously to the other applications until the importers are done, and ensures that all processes created in 8.7 are migrated and enhanced.
+
+The migration application can be found in the `camunda/bin/migrate` directory.
```
Is there any other step/information required here? Is there a specific point in the update process where this should be run (judging by the positioning, is it after the importers are done, but before you run the prefix migration?). It might be nice to call that out if so:
```suggestion
The migration application can be found in the `camunda/bin/migrate` directory. Ensure the application is run prior to updating any [custom prefixes](#update-any-custom-prefixes).
```

If there is no recommendation, we should state that.

ChrisKujawa replied:
The suggestion is not correct. The index prefix migration is only necessary if they customer used a custom prefix before and then IT MUST run as a first step. I just have this moved more to the back, as it is not always needed.

conceptualshark commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,140 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+This step is required if you use an installation with custom prefixes.
+:::
+
+The new harmonized index schema introduces one common index prefix. To update to Camunda 8.8, indices must first migrate to one common index, so the Camunda Exporter, REST API, and other consumers can be configured correctly.
```
I couldn't quite tell in this second sentence if "indices must first migrate to one common index" meant "existing indices using custom prefixes must be migrated to a common prefix" - is something along those lines more descriptive? It would be nice to be clear.

ChrisKujawa replied:
Makes sense, or sounds right.

conceptualshark commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,140 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+Ensure you have at least double the disk space free.
+:::
+
+:::note
+While we test and validate the provided migration script,
+it is recommended that you first back up the database, and test the migration script in a test environment first
+to ensure there are no unexpected issues.
+:::
```
Would it be a good practice to always test the script before running it, even after the migration is out of our own test/validate phrase? We could remove the first portion of the sentence if so.

ChrisKujawa replied:
Not fully sure, whether I understand what you mean. But I guess we can improve this note yes.

ChrisKujawa replied:
Improved https://github.com/camunda/camunda-docs/pull/5023/commits/52bf2add93d31034f16c11d2727d80be8dc108ed

conceptualshark commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,140 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+
+Complete the following steps to migrate any custom prefixes:
+
+1. Shut down your Camunda 8.7 cluster (for example, reduce the replicas to zero). No traffic can be executed against your to-be-migrated ES/OS prefixes.
+2. In the `camunda/bin` directory there is a `prefix-migration` script that will execute the prefix migration, which can be executed locally.
+   Pass the desired prefix using the `CAMUNDA_DATABASE_INDEXPREFIX` environment variable.
+
+   1. Make sure that your configuration in `camunda/config` has set the previous custom prefixes. Such that the prefix migration script can find them.
+   2. Alternatively, you can set them via env vars.
```
I had a hard time following the order of these steps - could you clarify if the following is true?
```suggestion
1. Shut down your Camunda 8.7 cluster (for example, reduce the replicas to zero). No traffic can be executed against your to-be-migrated ES/OS prefixes.
2. Set the `CAMUNDA_DATABASE_INDEXPREFIX` environment variable to your desired prefix.
3. Ensure that your configuration in `camunda/config` contains the previous custom prefixes, such that the prefix migration script can find them.
  - Alternatively, you can set them via environment variables:
4. In the `camunda/bin` directory, execute the `prefix-migration` script, which can be run locally.
```

"such that the prefix migration script can find them" wasn't clear to me, as it implies the script needs them formatted in a unique way. If the user has set custom prefixes, will these already be set? Are we having them verify? Otherwise, this reads like they might have to change something, and we should be specific if so.

ChrisKujawa replied:
>"such that the prefix migration script can find them" wasn't clear to me, as it implies the script needs them formatted in a unique way. If the user has set custom prefixes, will these already be set? Are we having them verify? Otherwise, this reads like they might have to change something, and we should be specific if so.

Agreed. it is a bit colloquial. IF the user has a custom prefix then it MUST be configured before in their configs, in their application.yaml. We need to have access to these values from the prefix migration application. We could be more specific and simply do: `SPRING_CONFIG_ADDITIONALLOCATION=/path/to/application.yaml ./camunda/bin/prefix-migration

conceptualshark commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
```diff
@@ -6,6 +6,140 @@ description: "Review which adjustments must be made to migrate from Camunda 8.7.
+Without these configuration changes, the update to 8.8 will fail.
+:::
+
+Given a new desired prefix of `some-new-prefix`:
+
+- The configured `CAMUNDA_OPERATE_ELASTICSEARCH_INDEX_PREFIX` must now be `some-new-prefix`.
+- The configured `CAMUNDA_TASKLIST_ELASTICSEARCH_INDEX_PREFIX` must now be `some-new-prefix`.
+- The prefix for the [Camunda exporter](/self-managed/zeebe-deployment/exporters/camunda-exporter.md) should be configured using `ZEEBE_BROKER_EXPORTERS_CAMUNDA_ARGS_INDEX_PREFIX` to `some-new-prefix`.
```
This section says configuration changes must be made, but these don't feel like instructions. If the user _must_ set this config following the completion of the script, I'd recommend the following: 
```suggestion
Given a new desired prefix of `some-new-prefix`:

- Set `CAMUNDA_OPERATE_ELASTICSEARCH_INDEX_PREFIX` to `some-new-prefix`.
- Set `CAMUNDA_TASKLIST_ELASTICSEARCH_INDEX_PREFIX` to `some-new-prefix`.
- Set `ZEEBE_BROKER_EXPORTERS_CAMUNDA_ARGS_INDEX_PREFIX` (the prefix for the new [Camunda exporter](/self-managed/zeebe-deployment/exporters/camunda-exporter.md) to `some-new-prefix`.
```

conceptualshark reviewed (commented):
I've cleaned up some of the language here, which should largely address @RomanJRW's comments, but also left a few questions regarding wording/steps where they were unclear to me. Some of them may just be my unfamiliarity with the subject, so just let me know if so!

ChrisKujawa reviewed (commented):
Thanks @conceptualshark do you have time for another review? I made several changes.

ChrisKujawa commented on docs/self-managed/operational-guides/update-guide/870-to-880.md:
````diff
@@ -4,7 +4,177 @@ title: Update 8.7 to 8.8
+/ sum(rate(operate_import_time_seconds_count{namespace=~"$namespace", partition=~"$partition", pod=~"$pod"}[$__rate_interval]))
+```
+
+### Data migration
````
@panagiotisgts I described a bit more the Migration application and the configs, could you please check this for correctness.

conceptualshark reviewed (approved):
I went ahead and reviewed + implemented the comments you approved, so this is looking good on my end! I'll go ahead and approve so when any remaining technical review is complete you will be good to merge. 👍

ChrisKujawa commented:
First of all thanks to all the reviewers! I think this brings a good foundation for our update guide related to the exporter project. I think we are close to done with this PR. I hope to get this merged soon (today?). I just need a new approval from someone.

@conceptualshark @akeller I had to merge `main` into this PR (I first tried to rebase, but it was a pain). It looks like this is common in the docs repo to use merge commits. Let me know if not. It was a bit pain, as the announcement pages have been changed. I hope I didn't break anything. In general I moved the Camunda Exporter announcement from the release notes to the actual announcements page under `Key changes`, as it is a fundamental and architectural change. Let me know if you disagree.

@panagiotisgts thanks for your input I incorporated your suggestions regarding config and metrics. 

Documentation is a living object, it should be seen as part of our code basis and product. If we change or refactor something we should consider adjusting it, please keep this in mind @aabouzaid @RomanJRW @sdorokhova mentioning you, as you likely will be in touch with the changes or related content, that is added with this PR. Especially mentioning this because I will switch projects, and might not be available for such changes. :)

panagiotisgts reviewed (commented):
LGTM for the Data Migration changes.

akeller commented:
> @conceptualshark @akeller I had to merge `main` into this PR (I first tried to rebase, but it was a pain). It looks like this is common in the docs repo to use merge commits. Let me know if not. It was a bit pain, as the announcement pages have been changed. I hope I didn't break anything. In general I moved the Camunda Exporter announcement from the release notes to the actual announcements page under `Key changes`, as it is a fundamental and architectural change. Let me know if you disagree.

@conceptualshark, do you have time to look at this today so we can merge? In general, we probably need to reinforce small, quick PRs as we continue to make significant docs changes for architectural streamlining.

conceptualshark reviewed (approved):
It looks like the build and link are fixed and I can't see anything else resulting from the merge - this seems to be in good shape to me! 👍 Let me know if you need any other feedback or review.
